### ISSUE TITLE
Documentation of ObservationRestTemplateCustomizer as replacement of MetricsRestTemplateCustomizer

### ISSUE THREAD

**Issue by MahatmaFatalError** [spring-projects/spring-boot#35918]
This issue https://github.com/spring-projects/spring-boot/issues/32484 with commit https://github.com/spring-projects/spring-boot/commit/eac50a8f0c449a6d756f0dd6e46cc67024708713#diff-4cdef5c5ee8ef10a5d751cbb07a000805bac055f8fa8a1620b0e219826e3eb74 removed `org.springframework.boot.actuate.metrics.web.client.MetricsRestTemplateCustomizer` and introduced `org.springframework.boot.actuate.metrics.web.client.ObservationRestTemplateCustomizer`. I assume the latter is a drop-in replacement, I don't know.

IMHO this breaking change should be documented, maybe https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide or https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Release-Notes.

**Comment by GirishSrinivas** [comment]
Hi,
we are doing the spring boot 3 migrations, and we are using the claas MetricsRestTemplateCustomizer for the restTemplate, we have not see any documentation about this change.
It would be greatly appreciated if you can point us to any documentation for the migration of  MetricsRestTemplateCustomizer to ObservationRestTemplateCustomizer

**Comment by Himanshu-PayPay** [comment]
Yes, please provide when we upgraded to 3.1.3 it started breaking. Also can you please share some sample example in order to send the http.client.requests metrices using  ObservationRestTemplateCustomizer

**Comment by bclozel** [comment]
Can you elaborate on why are you using this customizer in the first place? What type of customization are you applying here? I'm asking this because you are not required to define your own and this should be mostly a transparent change for Boot users.

**Comment by MahatmaFatalError** [comment]
@wilkinsona we used
```
metricsRestTemplateCustomizer.customize(restTemplate);
```
and now needed to replace it with 
```
observationRestTemplateCustomizer.customize(restTemplate);
```
to access `/actuator/metrics/http.client.requests` metrics.

**Comment by bclozel** [comment]
Given the use case, I'm not sure we need to add much more in the docs. You're not really supposed to use the customizers directly but rather [inject a `RestTemplateBuilder` in your application](https://docs.spring.io/spring-boot/docs/current/reference/html/io.html#io.rest-client.resttemplate). [I've added a note in the migration wiki still](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#deprecation-of-the-spring-boot-2x-instrumentation).
